teerex/src/lib.rs

```diff
@@ -446,6 +500,78 @@ impl<T: Config> Pallet<T> {
 			Err(<Error<T>>::RemoteAttestationTooOld.into())
 		}
 	}
+
+	fn check_block_and_get_latest_header(
+		shard_id: ShardIdentifier,
+		block_header: Header,
+		sender: &T::AccountId,
+		sender_index: u64,
+		latest_block_header: Header,
+	) -> Header {
+		// Confirm that the parent hash is the hash of the previous block.
+		// Block number 1 does not have a previous block, hence skip checking there.
+		if latest_block_header.hash() == block_header.parent_hash || block_header.block_number == 1
+		{
+			Self::confirm_sidechain_block(shard_id, block_header, sender, sender_index);
+			return block_header
+		}
+		latest_block_header
+	}
+
+	fn confirm_sidechain_block(
+		shard_id: ShardIdentifier,
+		block_header: Header,
+		sender: &T::AccountId,
+		sender_index: u64,
+	) {
+		<LatestSidechainBlockHeader<T>>::insert(shard_id, block_header);
+		<WorkerForShard<T>>::insert(shard_id, sender_index);
+		let block_hash = block_header.block_data_hash;
+		log::debug!(
+			"Proposed sidechain block confirmed with shard {:?}, block hash {:?}",
+			shard_id,
+			block_hash
+		);
+		Self::deposit_event(Event::ProposedSidechainBlock(sender.clone(), block_hash));
+	}
+
+	fn import_blocks_from_queue(
```

Comment:
I think `import` is not the correct term here? We finalize the blocks, not import.

Reply:
the plan is to extract this to a sidechain specific pallet (#49). This also resolves the confusion.